Clean how the project deal with DI (#78)
jolicode/elastically: README.md

## Demo

> If you are using Symfony, you can move to the Symfony [chapter](#usage-in-symfony)

Quick example of what the library do on top of Elastica:

```php
// Your own DTO, or one generated by Jane (see below)
class Beer
{
    public string $foo;
    public string $bar;
}

use JoliCode\Elastically\Factory;
use Elastica\Document;

> // New Factory object with new options

Reviewer: The logic is way different here. Before it was just an Elastica client compatible client.
Now we have to build a Factory, not a Client.

```suggestion
// Factory object with Elastica options + new Elastically options in the same array
```

$factory = new Factory([
    // Where to find the mappings
    Factory::CONFIG_MAPPINGS_DIRECTORY => __DIR__.'/mappings',
    // What object to find in each index
    Factory::CONFIG_INDEX_CLASS_MAPPING => [
        'beers' => Beer::class,
    ],
]);

> // Class to  the client

Reviewer: ```suggestion
// Custom Elastica compatible Client
```

> // Class to perform request
> $client = $factory->buildClient();

Reviewer: ```suggestion
// Class to perform request, same as the Elastica Client
$client = $factory->buildClient();
```

]
```

### `Factory::CONFIG_SERIALIZER` (optional)

> A `SerializerInterface` compatible object that will by used both on indexation.

Reviewer: ```suggestion
A `SerializerInterface` compatible object that will by used on indexation.
```

_Default to Symfony Serializer with Object Normalizer._

A faster alternative is to use Jane to generate plain PHP Normalizer, see below. Also we recommend [customization to handle things like Date](https://symfony.com/doc/current/components/serializer.html#normalizers).

### `Factory::CONFIG_DENORMALIZER` (optional)

> A `DenormalizerInterface` compatible object that will by used both on search.

Reviewer: ```suggestion
A `DenormalizerInterface` compatible object that will by used on search results to build your objects back.
```

If this option is not defined, the factory will fallback to
`Factory::CONFIG_SERIALIZER` option.

### `Factory::CONFIG_SERIALIZER_CONTEXT_BUILDER` (optional)

An instance of `ContextBuilderInterface` that build a serializer context from a
class name.

> If it is not defined, elastically, will use a `StaticContextBuilder` with the

Reviewer: ```suggestion
If it is not defined, Elastically, will use a `StaticContextBuilder` with the
```
